src/routes/our-music/AudioPlayer.svelte

```diff
@@ -0,0 +1,296 @@
+        let durMins = Math.floor(duration / 60);
+        let durSecs = Math.floor(duration - durMins * 60);
+        let durHours = Math.floor(duration / 3600);
+
+        if (currHours > 0) currMins -= currHours * 60;
+        if (durHours > 0) durMins -= durHours * 60;
+
+        if (currSecs < 10) currSecs = `0${currSecs}`;
+        if (durSecs < 10) durSecs = `0${durSecs}`;
+        if (currMins < 10) currMins = `0${currMins}`;
+        if (durMins < 10) durMins = `0${durMins}`;
+
+        if (currHours == 0) currTimeDisplay = `⏱ ${currMins}:${currSecs}`;
+        if (durHours == 0) totalTimeDisplay = `${durMins}:${durSecs} ⏱`;
+
+        if (currHours > 0)
+            currTimeDisplay = `⏱ ${currHours}:${currMins}:${currSecs}`;
+        if (durHours > 0)
+            totalTimeDisplay = `${durHours}:${durMins}:${durSecs} ⏱`;
+
+        if (ended) {
+            toggleTimeRunning();
+            playNextTrack();
+            updateTimeV2();
+        }
+    }
+
+    function updateTimeV2() {
+        let currMins = Math.floor(audio.currentTime / 60);
+        let currSecs = Math.floor(audio.currentTime - currMins * 60);
+        let currHours = Math.floor(audio.currentTime / 3600);
+
+        let durMins = Math.floor(audio.duration / 60);
+        let durSecs = Math.floor(audio.duration - durMins * 60);
+        let durHours = Math.floor(audio.duration / 3600);
+
+        if (currHours > 0) currMins -= currHours * 60;
+        if (durHours > 0) durMins -= durHours * 60;
+
+        if (currSecs < 10) currSecs = `0${currSecs}`;
+        if (durSecs < 10) durSecs = `0${durSecs}`;
+        if (currMins < 10) currMins = `0${currMins}`;
+        if (durMins < 10) durMins = `0${durMins}`;
+
+        if (currHours == 0) currTimeDisplay = `⏱ ${currMins}:${currSecs}`;
+        if (durHours == 0) totalTimeDisplay = `${durMins}:${durSecs} ⏱`;
+
+        if (currHours > 0)
+            currTimeDisplay = `⏱ ${currHours}:${currMins}:${currSecs}`;
+        if (durHours > 0)
+            totalTimeDisplay = `${durHours}:${durMins}:${durSecs} ⏱`;
+    }
+
+    const toggleTimeRunning = () => {
+        if (ended && trackIndex < audioData.length - 1) {
+            isPaused = true;
+            clearInterval(trackTimer);
+            // $: console.log(`Ended = ${ended}`);
+        } else if (ended) {
+            // this separation somehow works on handling end of playlist, so we just keep it that way
+            isPaused = true;
+            clearInterval(trackTimer);
+            // $: console.log(`Ended = ${ended}`);
+        } else {
+            trackTimer = setInterval(updateTime, 100);
+        }
+    };
+
+    // Controls
+    // $: console.log(`isPaused = ${isPaused}`);
+
+    const playPauseAudio = () => {
+        if (audio.paused) {
+            toggleTimeRunning();
+            audio.play();
+            isPaused = false;
+        } else {
+            toggleTimeRunning();
+            audio.pause();
+            isPaused = true;
+        }
+    };
```

Comment:
Also following my previous comments, this can be separated only if there are in fact two different dispatchers. The idea is since it works on the same button, the events should lie within the same method.

Reply:
I have a better suggestion, I can do the same like toggleTimeRunning where it becomes a wrapper of two small functions! Let me try this instead